packages/react-sdk/src/components/connection-state/ConnectionStateProvider.tsx

```diff
@@ -78,11 +84,24 @@ export const ConnectionStateProvider: React.FC<PropsWithChildren> = function ({
     setConnectionStateDialogOpen(true);
   }, [setConnectionStateDialogOpen]);
 
+  /**
+   * Monitor load snapshot state. Display dialog if snapshot load failed.
+   */
+  useEffect(() => {
+    if (!snapshotInfo.snapshotLoadFailed) {
+      // No load snapshot error - no dialog
+      setSnapshotLoadDialogOpen(false);
+      return;
+    }
+
+    setSnapshotLoadDialogOpen(true);
+  }, [snapshotInfo, snapshotLoadDialogOpen, setSnapshotLoadDialogOpen]);
```

Comment:
We can use `snapshotInfo.snapshotLoadFailed` in dependencies to save `useEffect` calls.